src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -284,41 +284,80 @@ def makeSubscriptions(self, workflow):
                                                     site=workflow['destination'])
-                    except Exception as ex:
-                        msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
-
-                    try:
-                        # In the message bellow we may want to put the list of datasets too
-                        msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
-                        self.logger.info(msg)
-                        ddmResult = self.ddm.makeRequest(ddmRequest)
-                    except Exception as ex:
-                        msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
+                        ddmReqList.append(ddmRequest)
+                except Exception as ex:
+                    msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
+
+                try:
+                    # In the message bellow we may want to put the list of datasets too
+                    msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
+                    self.logger.info(msg)
+                    ddmResultList = self.ddm.makeAggRequests(ddmReqList, aggKey='item')
+                except Exception as ex:
+                    msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
 
+                ddmStatusList = ['new', 'activated', 'completed', 'rejected', 'cancelled']
+                transferIDs = []
+                transferStatusList = []
+                for ddmResult in ddmResultList:
                     if 'data' in ddmResult.keys():
-                        self.docKeyUpdate(workflow,
-                                          transferStatus=deepcopy(ddmResult['data'][0]['status']),
-                                          transferIDs=deepcopy(ddmResult['data'][0]['request_id']))
-                        return workflow
-                    else:
-                        msg = "No data found in ddmResult for %s. Either dry run mode or " % workflow['RequestName']
-                        msg += "broken transfer submission to DDM. "
-                        msg += "ddmResult: \n%s" % pformat(ddmResult)
-                        self.logger.warning(msg)
+                        id = deepcopy(ddmResult['data'][0]['request_id'])
```

Comment:
isn't it an immutable object (either integer or string)? No need for deepcopy it (same for the line below)

Reply:
I just wanted to stay on the safe side - it shouldn't break if it is or not an immutable, though.